src/cookie.rs

```diff
@@ -0,0 +1,252 @@
+    pub(self) name: S,
+    pub(self) path: Option<S>,
+    pub(self) persistent: bool,
+    pub(self) secure: bool,
+    pub(self) value: S,
+}
+
+impl<S> Cookie<S>
+where
+    S: AsRef<str>,
+{
+    /// If any, returns the domain associated with this
+    /// cookie. E.g.: `Set-Cookie: name=value; Domain=.foo.com`.
+    pub fn domain(&self) -> Option<&str> {
+        self.domain.as_ref().map(|x| x.as_ref())
+    }
+
+    /// If any, returns the time this cookie expires.
+    ///
+    /// A value less than the current time means that the cookie has been expired.
+    pub fn expires_at(&self) -> Option<&Duration> {
+        self.expires_at.as_ref()
+    }
+
+    /// Converts itself into [`CookieBuilder`](CookieBuilder). Useful
+    /// to override any parameter in a flexible and readable way.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use tide::CookieBuilder;
+    ///
+    /// let _ = CookieBuilder::new()
+    ///     .name("foo")
+    ///     .path("/a-path")
+    ///     .build()
+    ///     .into_builder()
+    ///     .name("another-foo")
+    ///     .build();
+    /// ```
+    pub fn into_builder(self) -> CookieBuilder<S> {
+        CookieBuilder {
+            domain: self.domain,
+            expires_at: self.expires_at,
+            host_only: self.host_only,
+            http_only: self.http_only,
+            name: Some(self.name),
+            path: self.path,
+            persistent: self.persistent,
+            secure: self.secure,
+            value: self.value,
+        }
+    }
+
+    /// If this cookie should be used for a single and unique domain.
+    pub fn host_only(&self) -> bool {
+        self.host_only
+    }
+
+    /// If this cookie should be restrained to HTTP APIs only.
+    pub fn http_only(&self) -> bool {
+        self.http_only
+    }
+
+    /// The name that identifies this cookie. E.g.: `Set-Cookie: name=value`.
+    pub fn name(&self) -> &str {
+        self.name.as_ref()
+    }
+
+    /// If any, returns the path associated with this
+    /// cookie. E.g.: `Set-Cookie: name=value; Path=/`.
+    pub fn path(&self) -> Option<&str> {
+        self.path.as_ref().map(|x| x.as_ref())
+    }
+
+    /// If this cookie should not expire at the end of the current session.
+    pub fn persistent(&self) -> bool {
+        self.persistent
+    }
+
+    /// If this cookie should only be transmitted in HTTPS connections.
+    pub fn secure(&self) -> bool {
+        self.secure
+    }
+
+    /// The cookie's value. E.g.: `Set-Cookie: name=value`.
+    pub fn value(&self) -> &str {
+        self.value.as_ref()
+    }
+}
+
+/// Provides a handy interface to build a [`Cookie`](Cookie) readably.
+#[derive(Clone, Debug, Default)]
+pub struct CookieBuilder<S> {
+    domain: Option<S>,
+    expires_at: Option<Duration>,
+    host_only: bool,
+    http_only: bool,
+    name: Option<S>,
+    path: Option<S>,
+    persistent: bool,
+    secure: bool,
+    value: S,
+}
+
+impl<S> CookieBuilder<S>
```

Comment:
All of the methods on `CookieBuilder` should probably be tagged with `#[inline]`.